core/src/main/kotlin/gropius/dto/input/architecture/UpdateIMSInput.kt

```diff
@@ -0,0 +1,23 @@
+package gropius.dto.input.architecture
+
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.execution.OptionalInput
+import gropius.dto.input.common.JSONFieldInput
+import gropius.dto.input.common.UpdateNamedNodeInput
+import gropius.dto.input.common.ensureNoDuplicates
+import gropius.dto.input.ifPresent
+import gropius.dto.input.template.UpdateTemplatedNodeInput
+
+@GraphQLDescription("Input for the updateIMS mutation")
+class UpdateIMSInput(
+    @GraphQLDescription("Values for templatedFields to update")
+    override val templatedFields: OptionalInput<List<JSONFieldInput>>
```

Comment:
technically, no  
However, I want the user to not provide it (and not provide null) to have a clearer contract (basically do not use null as an alternative for not provided, as there are cases where it makes a difference)

Reply:
But what if one wants to provide it as gql variable or otherwise programmatically where he provides it but sets it to a variable value? Will this accept `undefined` as equal to not given?

And if you do `OptionalInput`s we should propably do it consistently everywhere (and note it somewhere in the docs that `null` is not equivalent to not given as this isn't represented in the gql schema)

Reply:
afaik it treats undefined as not given, and null as null  
however even if not, the user must have a way to distinguish it, as there are cases where it definitely will make a difference  

for the documentation, I would say that we put somewhere in our general documentation "for all mutations, null is only allowed where explicitly specified, for all queries, null is interpreted like not defined"

Reply:
Yep, sounds reasonable